### PR TITLE
Update warning on import when both Shapely 2 and PyGEOS are installed

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Trying to make it a bit more explicit about when and what will change.

xref https://github.com/geopandas/geopandas/issues/2691

### CODE REVIEW

File: geopandas/_compat.py [hunk 1]
```diff
@@ -137,16 +137,21 @@ def set_use_pygeos(val=None):
     if USE_PYGEOS and env_use_pygeos is None and SHAPELY_GE_20:
         warnings.warn(
             "Shapely 2.0 is installed, but because PyGEOS is also installed, "
-            "GeoPandas will still use PyGEOS by default for now. To force to use and "
-            "test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. "
+            "GeoPandas still uses PyGEOS by default. However, starting with the next "
+            "release (0.14), the default will switch to Shapely. To already force to "
```
Comment (brendan-ward) [inline geopandas/_compat.py:141]: Suggestion: `starting with the next release (0.14)` => `starting with version 0.14`?

Comment (brendan-ward) [inline geopandas/_compat.py:141]:
```suggestion
            "release (0.14), the default will switch to Shapely. To force to "
```

File: geopandas/_compat.py [hunk 2]
```diff
@@ -137,16 +137,21 @@ def set_use_pygeos(val=None):
...
+            "use Shapely 2.0 now, you can either uninstall PyGEOS or set the "
+            "environment variable USE_PYGEOS=0. "
             "You can do this before starting the Python process, or in your code "
             "before importing geopandas:"
             "\n\nimport os\nos.environ['USE_PYGEOS'] = '0'\nimport geopandas\n\n"
-            "In a future release, GeoPandas will switch to using Shapely by default. "
-            "If you are using PyGEOS directly (calling PyGEOS functions on geometries "
+            "In the next release, GeoPandas will switch to using Shapely by default, "
+            "even if PyGEOS is installed. If you are only having PyGEOS installed to "
```
Comment (brendan-ward) [inline geopandas/_compat.py:148]:
```suggestion
            "even if PyGEOS is installed. If you only have PyGEOS installed to "
```

**Review by brendan-ward — APPROVED** [review]
A few minor text suggestions but otherwise this is good to go; thanks @jorisvandenbossche

**Comment by jorisvandenbossche** [comment]
Failures are because of a recent pyogrio change.

**Pull request merged**